### ISSUE TITLE
Parallelize the calculation of p-values

### ISSUE THREAD

**Issue by jneuff** [blue-yonder/tsfresh#3]
Currently we calculate the p-values per feature in a [loop](https://github.com/blue-yonder/tsfresh/blob/master/tsfresh/feature_selection/feature_selector.py#L117):

``` python
for feature in df_features['Feature']:
        if target_is_binary:
            # Decide if the current feature is binary or not
            if len(set(X[feature].values)) == 2:
                df_features.loc[df_features.Feature == feature, "type"] = "binary"
                p_value = target_binary_feature_binary_test(X[feature], y, settings)
            else:
                df_features.loc[df_features.Feature == feature, "type"] = "real"
                p_value = target_binary_feature_real_test(X[feature], y, settings)
        else:
            # Decide if the current feature is binary or not
            if len(set(X[feature].values)) == 2:
                df_features.loc[df_features.Feature == feature, "type"] = "binary"
                p_value = target_real_feature_binary_test(X[feature], y, settings)
            else:
                df_features.loc[df_features.Feature == feature, "type"] = "real"
                p_value = target_real_feature_real_test(X[feature], y, settings)

        # Add p_values to df_features
        df_features.loc[df_features['Feature'] == feature, "p_value"] = p_value
```

This loop could be parallelized with joblib or something similar.

**Comment by nils-braun** [comment]
Why not just use multithreading Pools? Because the jobs should be rather independent and caching will probably not help much, as these calculation is mostly done only once (or did I miss something?)

**Comment by MaxBenChrist** [comment]
You are right, we can use processes, there is not need to use shared memory or objects. (I mean the target vector gets duplicated but that should not be much of a problem)

**Comment by MaxBenChrist** [comment]
@nils-braun implemented this in pull request #76
